dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/params/dimension/DimensionParamType.java

```diff
@@ -80,9 +80,14 @@ public enum DimensionParamType {
   private static List<String> parseDate(
       CommonRequestParams commonRequestParams, AnalyticsDateFilter analyticsDateFilter) {
 
-    return SetUtils.emptyIfNull(
-            analyticsDateFilter.getTrackedEntityExtractor().apply(commonRequestParams))
-        .stream()
+    Function<CommonRequestParams, java.util.Set<String>> extractor =
+        analyticsDateFilter.getTrackedEntityExtractor();
+
+    if (extractor == null) {
```

Comment:
Maybe, it's null because of the new date introduced?

```
  COMPLETED_DATE(
      TimeField.COMPLETED_DATE,
      EventsAnalyticsQueryCriteria::getCompletedDate,
      EnrollmentAnalyticsQueryCriteria::getCompletedDate,
      null),
  CREATED_DATE(
      TimeField.CREATED_DATE,
      EventsAnalyticsQueryCriteria::getCreatedDate,
      EnrollmentAnalyticsQueryCriteria::getCreatedDate,
      null);
```

Shouldn't we have the respective param for each new date, instead of providing `null`?